File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] sol = new int[nums.Length];
        int p = 1;
        for (int i = 0; i < nums.Length; i++) {
            sol[i] = p;
            p *= nums[i];
        }
        int s = 1;
        for (int i = nums.Length - 1; i >= 0; i--) {
            sol[i] *= s;
            s *= nums[i];
        }
        return sol;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        indexes = deque()

        sol = []

        for i, n in enumerate(nums):
            while indexes and nums[indexes[-1]] <= n:
                indexes.pop()
            indexes.append(i)

            if indexes[0] <= i - k:
                indexes.popleft()
            
            if i >= k - 1:
                sol.append(nums[indexes[0]]);
        
        return sol